map/routing_manager.cpp

```diff
@@ -702,6 +710,15 @@ bool RoutingManager::InsertRoute(Route const & route)
           FillTurnsDistancesForRendering(segments, subroute->m_baseDistance, subroute->m_turns);
           break;
         }
+      case RouterType::Helicopter:
+        {
+          subroute->m_routeType = df::RouteType::Helicopter;
+          subroute->m_headFakeDistance = -1.0f;
+          //Assuming that Helicopter line is shorter than 90°. Otherwise line tail would have a gray color.
+          subroute->m_tailFakeDistance = 90.0f;
```

Comment:
Add comment what do we want to achieve here?
Seems like m_headFakeDistance, m_tailFakeDistance are not supposed to modify outside.
They are for CreateDrapeSubroute internals.

Reply:
When I draw straight line with `route_dash.fsh.glsl`
<img alt="drape-schema-grey-out" src="https://github.com/organicmaps/organicmaps/assets/720808/beca1c0b-e7f2-48dc-b466-4d5b8e7990f0" width="300px"/>

Line tails color is changed to `u_fakeColor` after some distance. Previosly there was no such long lines. So I increased this distance to 90° in order to not change shader code:

<img alt="drape-schema" src="https://github.com/organicmaps/organicmaps/assets/720808/1a864ba5-1af4-44b6-9e3f-3c5d4f493139" width="300px"/>

Reply:
An idea came to my mind. Maybe I should change `u_fakeColor` value for Helicopter route?
This color is loaded in `RouteRenderer::RenderSubroute(...)` method:

```
df::GetColorConstant(kRouteFakeColor)
```

Is RouteRenderer better place to fix the color?

Reply:
Ah, I see ..
m_headFakeDistance and m_tailFakeDistance are about drawing grey (_fake_) lines when entering/leaving the normal route.

In helicopter router we don't need it, so better to set:
```
auto constexpr kBias = 1.0;
subroute->m_headFakeDistance = -kBias;
subroute->m_tailFakeDistance = totalLength + kBias;
```
Like it is done in CreateDrapeSubroute. Or if we call CreateDrapeSubroute, should debug why route parts are marked as _fake_.

Reply:
@vng  fixed `m_tailFakeDistance` in commit [4825fde](https://github.com/strump/organicmaps/commit/4825fde06f842a64fe119a23139ae43dee36a4c7)

I'm little worried that we have two special route types: transit and helicopter. I have to pass both flags `bool isTransit`, `bool isHelicopter` in couple of places. If we have any new such special case in future, let's think on better approach.

Reply:
Are they mutually exclusive? If yes, then an enum could be a better approach.